#!/usr/bin/env python
PACKAGE = "vision_motion_detection"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("skip", int_t, 0, "Number of frames to skip", 0, 0, 30)
gen.add("period", double_t, 0, "Output period in seconds", 0, 0, 5)

gen.add("blurring_size", int_t, 0, "Blurring size", 2, 0, 10)
gen.add("diff_threshold", int_t, 0, "The threshold for the difference between background and foreground", 30, 0, 255)
gen.add("dilate_size", int_t, 0, "Dilation size", 5, 0, 20)
gen.add("erode_size", int_t, 0, "Erosion size", 5, 0, 20)
gen.add("debug_view", bool_t, 0, "Display intermediate images", False)


exit(gen.generate(PACKAGE, "blob_extractor", "BlobExtractor"))